beater/v2_handler_test.go

```diff
@@ -81,3 +101,80 @@ func TestRequestDecoderError(t *testing.T) {
 
 	assert.Equal(t, http.StatusInternalServerError, w.Code, w.Body.String())
 }
+
+func TestRequestIntegration(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		code int
+		path string
+	}{
+		{"Success", http.StatusAccepted, "../testdata/intake-v2/errors.ndjson"},
+		{"InvalidEvent", http.StatusBadRequest, "../testdata/intake-v2/invalid-event.ndjson"},
+		{"InvalidJSONEvent", http.StatusBadRequest, "../testdata/intake-v2/invalid-json-event.ndjson"},
+		{"InvalidJSONMetadata", http.StatusBadRequest, "../testdata/intake-v2/invalid-json-metadata.ndjson"},
+		{"InvalidMetadata", http.StatusBadRequest, "../testdata/intake-v2/invalid-metadata.ndjson"},
+		{"InvalidMetadata2", http.StatusBadRequest, "../testdata/intake-v2/invalid-metadata-2.ndjson"},
+		{"UnrecognizedEvent", http.StatusBadRequest, "../testdata/intake-v2/unrecognized-event.ndjson"},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			b, err := loader.LoadDataAsBytes(test.path)
+			require.NoError(t, err)
+			bodyReader := bytes.NewBuffer(b)
+
+			req := httptest.NewRequest("POST", "/v2/intake", bodyReader)
+			req.Header.Add("Content-Type", "application/x-ndjson")
+
+			w := httptest.NewRecorder()
+
+			c := defaultConfig("7.0.0")
+			report := func(context.Context, publish.PendingReq) error {
+				return nil
+			}
+			handler := (&v2BackendRoute).Handler(c, report)
+
+			handler.ServeHTTP(w, req)
+
+			assert.Equal(t, test.code, w.Code, w.Body.String())
+			if test.code == http.StatusAccepted {
+				assert.Equal(t, 0, w.Body.Len())
+			} else {
+				body := w.Body.Bytes()
+				approveResultBody(t, "TestRequestIntegration"+test.name, body)
+			}
+		})
+	}
+}
+
+func TestReportingProblemsIntegration(t *testing.T) {
```

Comment:
I'd appreciate to get rid of the duplicated code in `TestRequestIntegration` and `TestReportingProblemsIntegration`.

Reply:
good tip!